### DISCUSSION TITLE
Pi Pico GP15 pin - to be avoided, as Adafruit said, or is it okay with this core?

### DISCUSSION THREAD

**Discussion by Andy2No** [earlephilhower/arduino-pico#364] [Q&A] [answered]
On the pinout for the Raspberry Pi Pico from Adafruit, they've marked pin GP15 as "Do not use".  I found a discussion about why, but I'm not clear if that only applies to using Micro Python / Circuit Python, or it it applies in general.

I found this thread on eevblog: https://www.eevblog.com/forum/microcontrollers/gp15-on-raspberry-pi-pico/

Which links to this official github issue thread: https://github.com/adafruit/circuitpython/issues/4034

**Comment by earlephilhower** [comment, marked as answer]
I wasn't aware of this.  From the descriptions this is a hardware issue with a stepping of the RP2040 chip, so this core is subject to the same limitations.

I've not actually noticed any issues, but I have A(0 or 1) steppings of the chip since I got them on release day.  If you're unsure of the stepping, then I'd stay away from pin 15 for safety.

Good luck!

**Reply by dhalbert** [reply]
This should be ameliorated by https://github.com/raspberrypi/pico-sdk/pull/80, which is now in the pico-sdk. The avoidance of pin 15 was done early on, and things should be better now.

**Reply by dhalbert** [reply]
Note that we suggest avoiding it only for earlier versions of CircuitPython: https://learn.adafruit.com/getting-started-with-raspberry-pi-pico-circuitpython/faq-and-troubleshooting#should-i-use-pin-gp15-3082470-3.

**Reply by earlephilhower** [reply]
Awesome, thanks for the clarification.  We're running the SDK 1.2 here, so those patches are already in place.

**Reply by Andy2No** [reply]
Thanks, dhalbert and earlephilhower.
